### ISSUE TITLE
Regression: missing/unknown rules are not logged anymore

### ISSUE THREAD

**Issue by Bertk** [wenns/sonar-cxx#361]
In the past a missing definition of a  rule in sensor profile could be identified bcause it was logged:
```
CxxUtils.LOG.warn("Cannot find the rule {}, skipping violation", ruleId);
```
The refactored version of CxxReportSensor.java does not log this warning anymore.

**Comment by jmecosta** [comment]
i think now you get  a analysis failure, at least with latest.

**Comment by wenns** [comment]
I also do not see such a warning with 4.5.1 and I think this is bad. I'll have a look...

**Comment by wenns** [comment]
@Bertk: I consider this an important feature, so I also added a behave test for it

**Comment by jmecosta** [comment]
pretty sure i got a build error because of this... will try again....

**Comment by jmecosta** [comment]
there, these are custom rules created in 4.5.1... 

16:04:10.078 DEBUG - Release semaphore on project : org.sonar.api.resources.Project@e3582d[id=1,key=googletest_bullseye_vs_project,qualifier=TRK], with key batch-googletest_bullseye_vs_project
INFO: ------------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
Total time: 4.486s
Final Memory: 21M/723M
INFO: ------------------------------------------------------------------------
ERROR: Error during Sonar runner execution
org.sonar.runner.impl.RunnerException: Unable to execute Sonar
	at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:91)
	at org.sonar.runner.impl.BatchLauncher$1.run(BatchLauncher.java:75)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.sonar.runner.impl.BatchLauncher.doExecute(BatchLauncher.java:69)
	at org.sonar.runner.impl.BatchLauncher.execute(BatchLauncher.java:50)
	at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:102)
	at org.sonar.runner.api.Runner.execute(Runner.java:90)
	at org.sonar.runner.Main.executeTask(Main.java:70)
	at org.sonar.runner.Main.execute(Main.java:59)
	at org.sonar.runner.Main.main(Main.java:41)
Caused by: org.sonar.api.utils.SonarException: Cannot feed the data into sonar, details: 'The rule 'other:cpplint_tekla_custom_include_files_0' does not exist.'
	at org.sonar.plugins.cxx.utils.CxxReportSensor.analyse(CxxReportSensor.java:127)
	at org.sonar.batch.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:79)
	at org.sonar.batch.phases.SensorsExecutor.execute(SensorsExecutor.java:70)
	at org.sonar.batch.phases.PhaseExecutor.execute(PhaseExecutor.java:119)
	at org.sonar.batch.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:194)
	at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:92)
	at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:77)
	at org.sonar.batch.scan.ProjectScanContainer.scan(ProjectScanContainer.java:233)
	at org.sonar.batch.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:228)
	at org.sonar.batch.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:221)
	at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:92)
	at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:77)
	at org.sonar.batch.scan.ScanTask.scan(ScanTask.java:64)
	at org.sonar.batch.scan.ScanTask.execute(ScanTask.java:51)
	at org.sonar.batch.bootstrap.TaskContainer.doAfterStart(TaskContainer.java:125)
	at org.sonar.api.platform.ComponentContainer.startComponents(ComponentContainer.java:92)
	at org.sonar.api.platform.ComponentContainer.execute(ComponentContainer.java:77)
	at org.sonar.batch.bootstrap.BootstrapContainer.executeTask(BootstrapContainer.java:173)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:95)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:67)
	at org.sonar.runner.batch.IsolatedLauncher.execute(IsolatedLauncher.java:45)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:87)
	... 9 more
Caused by: The rule 'other:cpplint_tekla_custom_include_files_0' does not exist.

**Comment by wenns** [comment]
You get this analysis error because of this commit?

**Comment by jmecosta** [comment]
nop, just making the point that when i run before i did not get no warning, i only get a analysis error. with this commit it should now pass and the warning logged.. so same behavior as in previous versions

**Comment by Bertk** [comment]
I will test the solution with Cppcheck V1.68 and see whether I find now logging information *"Could not add the issue, details:"*. I will post the results next Monday.
By the way, I could not identify new messages comparing error list of Cppcheck V1.67 with V1.68 but my experience told me to better check it with a larger source code base ;-)

**Comment by Bertk** [comment]
I could not find any message "Could not add the issue, details:" in the log file. In other words - no update of current rules repository for CppCheck V1.68 required.
